igvm/vm.py

```diff
@@ -1106,6 +1106,34 @@ def aws_sync(self) -> dict:
 
         return sync_values
 
+    def is_aws_image_golden(self) -> bool:
+        """Return whether the VM images is golden (created by us) or not"""
+
+        response = self.ec2c.describe_images(
+            ImageIds=[
+                self.dataset_obj['aws_image_id'],
+            ]
+        )
+
+        tags = response['Images'][0].get('Tags')
+
+        if not tags:
+            log.warning(
+                "Couldn't find any tags in provided image. "
+                "Assuming it is a not self build image"
+            )
+            return False
+
+        for tag in tags:
+            if tag['Key'] == 'golden_image':
+                log.info(
```

Comment:
This log message does not belong here because the purpose of the function is to determine whether the VM was built on a golden image. It's purpose is not to control the Puppet behavior